lib/rucio/db/sqla/__init__.py

```diff
@@ -36,3 +39,21 @@ def filter_thread_work(session, query, total_threads, thread_id, hash_variable=N
             else:
                 query = query.filter(text('mod(abs((\'x\'||md5(%s::text))::bit(32)::bigint), %s) = %s' % (hash_variable, total_threads, thread_id)))
     return query
+
+
+def filter_thread_work_where(session: "Session", stmt: "Select", total_threads: Optional[int], thread_id: Optional[int], hash_variable=None):
```

Comment:
Duplicating this function is not needed. where is a synonym to filter. Both for the query API and the statement api.

Reply:
Removed this in new commit